runtime/kubernetes/container.go

```diff
@@ -53,17 +53,9 @@ func (c *client) InspectContainer(ctx context.Context, ctn *pipeline.Container)
 
 		// avoid a panic if the build ends without terminating all containers
 		if cst.State.Terminated == nil {
-			for _, container := range pod.Spec.Containers {
```

Comment:
I realized that the image is already in `cst` so we don't need to look it up. That simplified this logic quite a bit.